ubtue_Brill.js

```diff
@@ -131,16 +131,26 @@ function postProcess(doc, item) {
 	} else {
 		item.date;
 	}
-
 	//scrape ORCID from website
 	let authorSectionEntries = doc.querySelectorAll('.text-subheading span');
+	let foundOrcid = false;
 	for (let authorSectionEntry of authorSectionEntries) {
 		let authorInfo = authorSectionEntry.querySelector('.c-Button--link');
 		let orcidHref = authorSectionEntry.querySelector('.orcid');
 		if (authorInfo && orcidHref) {
 			let author = authorInfo.childNodes[0].textContent;
 			let orcid = orcidHref.textContent.replace(/.*(\d{4}-\d+-\d+-\d+x?)$/i, '$1');
 			item.notes.push({note: "orcid:" + orcid + ' | ' + author});
+			foundOrcid = true;
+		}
+	}
+	if (!foundOrcid) authorSectionEntries = ZU.xpath(doc, '//div[@class="contributor-details"]');
+	for (let authorSectionEntry of authorSectionEntries) {
+		let authorInfo = ZU.xpathText(authorSectionEntry, './/*[@class="contributor-details-link"][1]')
+		let orcidHref = authorSectionEntry.querySelector('.orcid');
+		if (authorInfo && orcidHref) {
+			let orcid = orcidHref.textContent.replace(/.*(\d{4}-\d+-\d+-\d+x?)$/i, '$1');
+			item.notes.push({note: "orcid:" + orcid + ' | ' + authorInfo});
```

Comment:
Please preferably factor out the ORCID extraction to a separate function and and avoid the duplication of the author and ORCID extraction in lines (c.f. 141-143 and 150-153). Moreover the `foundOrcid` does not prevent rerunning the extraction in lines 148-153.

Reply:
The duplication of the author is due to the html structure, where the orcid section is repeated. But I could solve the problem, because the descendant elements of both orcid sections are different.